packages/core/src/indexing-store/cache.test.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  setupCleanup,
+  setupCommon,
+  setupDatabaseServices,
+  setupIsolatedDatabase,
+} from "@/_test/setup.js";
+import { onchainTable } from "@/drizzle/onchain.js";
+import { ZERO_CHECKPOINT_STRING } from "@/utils/checkpoint.js";
+import { zeroAddress } from "viem";
+import { beforeEach, expect, test } from "vitest";
+import { createIndexingCache } from "./cache.js";
+import { createHistoricalIndexingStore } from "./historical.js";
+
+beforeEach(setupCommon);
+beforeEach(setupIsolatedDatabase);
+beforeEach(setupCleanup);
+
+test("flush", async (context) => {
+  const schema = {
+    account: onchainTable("account", (p) => ({
+      address: p.hex().primaryKey(),
+      balance: p.bigint().notNull(),
+    })),
+  };
+
+  const { database } = await setupDatabaseServices(context, {
+    schemaBuild: { schema },
+  });
+
+  const indexingCache = createIndexingCache({
+    common: context.common,
+    database,
+    schemaBuild: { schema },
+    checkpoint: ZERO_CHECKPOINT_STRING,
+  });
+
+  const indexingStore = createHistoricalIndexingStore({
+    common: context.common,
+    database,
+    schemaBuild: { schema },
+    indexingCache,
+    db: database.qb.drizzle,
+  });
```

Comment:
One idea - you could consider an API that looks like this to manage the `tx` dep injection:
```ts
  const indexingStore = createHistoricalIndexingStore({
    common: context.common,
    database,
    schemaBuild: { schema },
    indexingCache,
    // db: database.qb.drizzle,
  });

await indexingStore.update(database.qb.drizzle, schema.account, { pk: value }).set(...)
// or, in a transaction
await indexingStore.update(tx, schema.account, { pk: value }).set(...)
```

Reply:
Not sure I follow, this indexing store almost directly makes it way into user code